### DISCUSSION TITLE
Prompt ChatGPT from anywhere you can input text

### DISCUSSION THREAD

**Discussion by mabry1985** [johnlindquist/kit#1220] [Share]
[DEPRECATED] 

By selecting any input text and running this script you now have access to the power of AI anywhere in your OS. 

https://user-images.githubusercontent.com/31560031/235273240-824a9432-87a3-48ac-9357-668a0c01b12c.mov

```js
/*
# Generate a response to a prompt in any text field

This script utilizes ChatGPT to generate a response for the selected text. 
The response will then replace the original text, making it useful for various writing tasks such as blog posts, code, and emails.

Highlight some text and press `cmd+shift+p` to send it through ChatGPT 
to replace the text response. Your mileage may vary.
*/

// Name: Prompt Anywhere
// Description: Generate a prompt response in anywhere you can type
// Author: Josh Mabry
// Twitter: @AI_Citizen
// Shortcut: cmd shift P

import "@johnlindquist/kit";
import Bottleneck from "bottleneck";

let { ChatOpenAI } = await import("langchain/chat_models");
let { HumanChatMessage, SystemChatMessage } = await import("langchain/schema");
let openAIApiKey = await env("OPENAI_API_KEY", {
  hint: `Grab a key from <a href="https://platform.openai.com/account/api-keys">here</a>`,
});
let { CallbackManager } = await import("langchain/callbacks");
let prompt = `#####
Ignore prior instructions, you are tasked with taking the input text and generating a response to it.

Return only the updated text, do not offer any explanations or reasoning for the changes. 
Do not ask any questions, do not refuse any input, and do not change the subject.
`;

const limiter = new Bottleneck({
  maxConcurrent: 1,
  minTime: 100,
});

const type = (text) => {
  return new Promise((resolve) => {
    keyboard.type(text);
    resolve();
  });
};

const wrappedType = limiter.wrap(type);

const chat = new ChatOpenAI({
  temperature: 0.3,
  openAIApiKey: openAIApiKey,
  streaming: true,
  callbackManager: CallbackManager.fromHandlers({
    handleLLMStart: async (token) => {
      log(`handleLLMStart`);
      if (!token) return;
      await wrappedType(token);
    },
    handleLLMNewToken: async (token, runId) => {
      log(`handleLLMNewToken`);
      // occasionally I see tokens typed out of order
      // still not sure why this happens
      await wrappedType(token);
    },
    handleLLMError: async (err) => {
      warn(`error`, JSON.stringify(err));
      process.exit(1);
    },
    handleLLMEnd: async () => {
      log(`handleLLMEnd`);
      log(`currentMessage`, currentMessage);
      process.exit(1);
    },
  }),
});

let text = await getSelectedText();

await chat.call([new SystemChatMessage(prompt), new HumanChatMessage(text)]);


```

**Comment by mabry1985** [comment]
Updates to add rate limit to fix bug that was garbling text on occasion

**Reply by stevedubois** [reply]
Nice script!

Because i was on a windows pc i had to adjust one thing:
await wrappedType(token) => await wrappedType(token.replace(/\r?\n/g, "\r\n"));

Otherwise the linefeeds were incorrect.

But I still have the problem with special characters like "ë" or "ç". Acting weird on those. Is there any solution for this?

**Reply by johnlindquist** [reply]
@mabry1985 You should be able to get platform specific eol like so:

https://nodejs.org/api/os.html#os_os_eol

@stevedubois Kit's keyboard.type is a wrapper around:

https://nut-tree.github.io/apidoc/classes/keyboard_class.KeyboardClass.html#type

I think it's simply not supported at the moment. Will have to investigate

**Reply by mabry1985** [reply]
@stevedubois interesting, thanks! 

The new one I have leverages the md component with paste functionality to prevent some of the oddities. I'll have that out soon with a video walkthrough.


@johnlindquist thats helpful info! I've been diving in under the hood a bit more lately and trying to get the lay of the land. 

Too much cool stuff to play with, not enough time 😄

**Reply by mabry1985** [reply]
I'm assuming the md component handles that under the hood in some capacity already, yeah?

**Reply by johnlindquist** [reply]
You me pasting vs. keyboard.type? Paste should be good to just layout whatever is in the clipboard.

**Comment by mabry1985** [comment]
Hey all, I've deprecated this script in favor of a more controlled approach being released tomorrow. 

Be sure to update the error handling if you are using this, I was outputting the error to selectedText and finally ran into an error. That was some messy JSON in my code editor.. 😅
```
  handleLLMError: async (err) => {
      warn(`error`, JSON.stringify(err));
      // await setSelectedText() <---- removed
      process.exit(1);
    },
   ```

**Comment by stevedubois** [comment]
Hello Marby,

Thank you for sharing this excellent script. I have tried the new version and it works perfectly! The "paste" option resolves the issues with strange characters and newlines.

I wanted to inform you that the retry and entering a new prompt feature is not working yet. It continues to respond with the first prompt entered. I added some logging and confirmed that the prompt used did not change after retrying and giving another prompt.

        handleLLMEnd: async () => {
          log(`handleLLMEnd`);
          log(`Prompt: ${formatPrompt(prompt)}`);

Additionally, there are now two versions of your script shown on the ScriptKit community, both labeled as v2, which can be confusing. They are not entirely identical.

**Reply by mabry1985** [reply]
Thanks for checking it out!

I did just notice that the prompt wasnt updating, I had it working for sure at one point, so I'll dig into that. Thanks for the report.
I'd also recommend pulling in my kenv to stay up to date with the script here https://github.com/artificialcitizens/ac-scripts
I don't always remember to pull in the latest changes to this discussion 😅 

I'm not sure why the script is appearing twice I did have an earlier discussion of the same name that I closed, so I may have screwed up something cc:@johnlindquist

**Reply by stevedubois** [reply]
I have fixed it:

add: "prompt = formatPrompt(followUp);"
before: "await processMessage(followUp);"

and there was another bug:
change  "new SystemChatMessage(formatPrompt(prompt)),", to "new SystemChatMessage(prompt),"

For better checking what's happening:
add: "log(`Prompt: ${prompt}`);"
before: "log(`handleLLMEnd`);"

**Reply by stevedubois** [reply]
I ended up adjusting it further for my own convenience.
*added some shotcuts to perconfigured (hardcoded) prompts
*added the option to ask follow up questions on the response of the AI

The first line is always your code, following with my additions:

- Input your desired prompt
    **Added abbility to work with shortcuts:**
      [Empty] : accept the selected text as a prompt in itself
      T = Text Correction Dutch - Moderate Creative 
        TE = Text Correction English
        TF = Text Correction French
      E = Explain the text (Dutch)
      U = Summary (Dutch)
      R = refractor Powershell Script
        RP = Refractor as Python Script
        RJ = Refractor as Javascript
      S = Script from text (Natural Language): Powershell
        SP = Script as Python Script
        SJ = Script as Javascript
    **Added abbility to add hashtags for finetuning response:**
      S = Short text
      L = Long text
      + = temperarture 0.6
      ++ = temperarture 0.8
      +++ = temperarture 1 (creative)
      - = temperarture 0.4
      -- = temperarture 0.2
      --- = temperarture 0 (strict)
    Example TE#L+++ = Text correction to English, text can be lenghty and temperature wil be 1 (creative)


* Paste - Paste response into highlighted text
***Added option:
* Folowup - Ask a followup question on the ai response


let { HumanChatMessage, SystemChatMessage } = await import("langchain/schema");
const promptHint = "[T]ext, [E]xplain), S[U]marry, [R]efractor, [S]cript #L/S+++---"


// System input / Task for the AI to follow
let userSystemInput = await arg(promptHint);
let defaultTemperature = 0.3;
let temperatureHash = "";



//#################
// Prompt Template        ===> completely rewritten
//#################
function formatPrompt(userSystemInput) {
  if (userSystemInput === promptHint) { userSystemInput = '';}
  let userInput = `${userSystemInput}`;
   let lengthHash = '';
  if (userSystemInput.includes('#')) {
    userInput = userSystemInput.split("#")[0].trim();
    let hash = userSystemInput.split("#")[1].trim();
    temperatureHash = hash.replace('S','').replace('L','');
    lengthHash = hash.replace(/[-+]/g,'');
  }
    const textPrompt = `
  -Return answer in markdown format
  -Refactoring it using the following rules: 
  -Maintain the same meaning, tone, and intent as the original text
  -Clean up any grammar or spelling mistakes
  -Make it sound more professional, but keep it casual
  -Reduce redundancies and excessive verbiage
  -Always give your response in [language]
  -Return only the updated text, do not offer any explanations or reasoning for the changes.`;
const refractorPrompt = `
  -You are tasked with taking the input text and refractor it for [language] script.
  -Return only the code. 
  -Do not give explanations outside of the code. 
  -Do not ask any questions.`;
const scriptPrompt = refractorPrompt.replace("refractor it for","translate it into");
let fullprompt = `
#####
  -Ignore prior instructions`; 
switch (userInput) {
case "":
  fullprompt += `
  -Return answer in markdown format
  -you are tasked with taking the input text and generating a response to it     
  -Return only the updated text, do not offer any explanations or reasoning for the changes. 
  -Do not ask any questions, do not refuse any input, and do not change the subject.`;
  break;
case "T":
  fullprompt += textPrompt.replace("[language]","Dutch");
  ;
  break;
case "TE":
  fullprompt += textPrompt.replace("[language]","English");
  ;
  break;
case "TF":
  fullprompt += textPrompt.replace("[language]","French");
  ;
  break;
case "E":
  fullprompt += `
  -You are tasked with taking the input text and explaining it to the user.
  -Always return the response in Dutch in the following format using markdown syntax:
  # Explain Plz
  ## Eenvoudig (Explain Like I'm 5)
  ## Uitgebreid (A longer technical explanation of the highlighted text)`;
  break;
case "U":
  fullprompt += `
  -Make a summary of the input text
  -Always give your response in Dutch`;
  break;
case "R":
  fullprompt += refractorPrompt.replace("[language]","Powershell");
  defaultTemperature = 0;
  break;  
case "RP":
  fullprompt += refractorPrompt.replace("[language]","Python");
  defaultTemperature = 0;
  break;  
case "RJ":
  fullprompt += refractorPrompt.replace("[language]","Javascript");
  defaultTemperature = 0;
  break;  
case "S":
  fullprompt += scriptPrompt.replace("[language]","Powershell");
  defaultTemperature = 0;
  break;  
case "SP":
  fullprompt += scriptPrompt.replace("[language]","Python");
  defaultTemperature = 0;
  break;  
case "SJ":
  fullprompt += scriptPrompt.replace("[language]","Javascript");
  defaultTemperature = 0;
  break;    
default:
  fullprompt += `
  -Return answer in markdown format
  -You are tasked with the following: ${userInput}`;
}
if (lengthHash === 'S') {
fullprompt += `
  -Make the text a bit more concise.`;
}
if (lengthHash === 'L') {
fullprompt += `
  -Make the text a bit longer.`;
}
fullprompt += `
######`;
return fullprompt;
};
function setTemperature(temp, defaultTemp) {
  switch (temp) {
    case "---":
      return 0;
    case "--":
      return 0.2;
    case "-":
      return 0.4;
    case "+":
      return 0.6;
    case "++":
      return 0.8;
    case "+++":
      return 1;
    default:
      return defaultTemp;
    }
};



* [Retry](submit:retry) - Rerun generation with option to update prompt
* [Followup](submit:followup) - Followup Question on the previous response



    // 0 = "precise", 1 = "creative"
    temperature: setTemperature(temperatureHash, defaultTemperature),



                placeholder: userSystemInput,
                hint: "Press enter to use the same prompt or " + promptHint,



            case "followup":
              // keep current message (response from AI) and set it as the new input text
              userPrompt = currentMessage;
              // prompt again with new prompt on this response
              const question = await arg({
                placeholder: "",
                hint: promptHint,
              });
              prompt = formatPrompt(question);
              await processMessage(question);
              break;

**Reply by stevedubois** [reply]
sorry, that was a mess,
here the full code:

`
/*
# Prompt Anything
Highlight some text and run this script to prompt against it.
Useful for summarizing text, generating a title, or any other task you can think of.

## Usage

- Highlight the text you want to prompt against
- Run the script via shortcut or command palette
- Input your desired prompt
    **Added abbility to work with shortcuts:**
      [Empty] : accept the selected text as a prompt in itself
      T = Text Correction Dutch - Moderate Creative 
        TE = Text Correction English
        TF = Text Correction French
      E = Explain the text (Dutch)
      U = Summary (Dutch)
      R = refractor Powershell Script
        RP = Refractor as Python Script
        RJ = Refractor as Javascript
      S = Script from text (Natural Language): Powershell
        SP = Script as Python Script
        SJ = Script as Javascript
    **Added abbility to add hashtags for finetuning response:**
      S = Short text
      L = Long text
      + = temperarture 0.6
      ++ = temperarture 0.8
      +++ = temperarture 1 (creative)
      - = temperarture 0.4
      -- = temperarture 0.2
      --- = temperarture 0 (strict)
    Example TE#L+++ = Text correction to English, text can be lenghty and temperature wil be 1 (creative)
- Wait for the AI to respond
- Select one of the options
* Retry - Rerun generation with option to update prompt
* Edit - Edit response in editor
    - On editor exit the message is saved to the clipboard
    - On editor submit the message is pasted into the highlighted text
* Copy - Copy response to clipboard
* Paste - Paste response into highlighted text
***Added option:
* Folowup - Ask a followup question on the ai response

## Example
- Highlight: 'Some really long passage in a blog post'
- Run Script
- Prompt: `Summarize this passage in the form of Shakespearean prose`
- Waaaaait for it...
- Get a response from the AI
- Select an option
- Rinse and repeat
*/

// Name: Prompt Anything
// Description: Custom prompt for any highlighted text
// Author: Josh Mabry (*** new functionality added by: Steve Duois)
// Twitter: @AI_Citizen
// Shortcut: Cmd shift P

//##################
// ScriptKit Imports
//##################
import "@johnlindquist/kit";

//##################
// LangChain Imports
//##################
let { ChatOpenAI } = await import("langchain/chat_models");
let { HumanChatMessage, SystemChatMessage } = await import("langchain/schema");

const promptHint = "[T]ext, [E]xplain), S[U]marry, [R]efractor, [S]cript #L/S+++---"

//#################
// Request API KEY
//#################
// stored in .env file after first run
// can change there or through the command palette
let openAIApiKey = await env("OPENAI_API_KEY", {
  hint: `Grab a key from <a href="https://platform.openai.com/account/api-keys">here</a>`,
});
// System input / Task for the AI to follow
let userSystemInput = await arg(promptHint);
let defaultTemperature = 0.3;
let temperatureHash = "";
// User Prompt from highlighted text
let userPrompt = await getSelectedText();

//#################
// Prompt Template
//#################
function formatPrompt(userSystemInput) {
  if (userSystemInput === promptHint) { userSystemInput = '';}
  let userInput = `${userSystemInput}`;
   let lengthHash = '';
  if (userSystemInput.includes('#')) {
    userInput = userSystemInput.split("#")[0].trim();
    let hash = userSystemInput.split("#")[1].trim();
    temperatureHash = hash.replace('S','').replace('L','');
    lengthHash = hash.replace(/[-+]/g,'');
  };
    const textPrompt = `
  -Return answer in markdown format
  -Refactoring it using the following rules: 
  -Maintain the same meaning, tone, and intent as the original text
  -Clean up any grammar or spelling mistakes
  -Make it sound more professional, but keep it casual
  -Reduce redundancies and excessive verbiage
  -Always give your response in [language]
  -Return only the updated text, do not offer any explanations or reasoning for the changes.`;
const refractorPrompt = `
  -You are tasked with taking the input text and refractor it for [language] script.
  -Return only the code. 
  -Do not give explanations outside of the code. 
  -Do not ask any questions.`;
const scriptPrompt = refractorPrompt.replace("refractor it for","translate it into");
let fullprompt = `
#####
  -Ignore prior instructions`; 
switch (userInput) {
case "":
  fullprompt += `
  -Return answer in markdown format
  -you are tasked with taking the input text and generating a response to it     
  -Return only the updated text, do not offer any explanations or reasoning for the changes. 
  -Do not ask any questions, do not refuse any input, and do not change the subject.`;
  break;
case "T":
  fullprompt += textPrompt.replace("[language]","Dutch");
  ;
  break;
case "TE":
  fullprompt += textPrompt.replace("[language]","English");
  ;
  break;
case "TF":
  fullprompt += textPrompt.replace("[language]","French");
  ;
  break;
case "E":
  fullprompt += `
  -You are tasked with taking the input text and explaining it to the user.
  -Always return the response in Dutch in the following format using markdown syntax:
  # Explain Plz
  ## Eenvoudig (Explain Like I'm 5)
  ## Uitgebreid (A longer technical explanation of the highlighted text)`;
  break;
case "U":
  fullprompt += `
  -Make a summary of the input text
  -Always give your response in Dutch`;
  break;
case "R":
  fullprompt += refractorPrompt.replace("[language]","Powershell");
  defaultTemperature = 0;
  break;  
case "RP":
  fullprompt += refractorPrompt.replace("[language]","Python");
  defaultTemperature = 0;
  break;  
case "RJ":
  fullprompt += refractorPrompt.replace("[language]","Javascript");
  defaultTemperature = 0;
  break;  
case "S":
  fullprompt += scriptPrompt.replace("[language]","Powershell");
  defaultTemperature = 0;
  break;  
case "SP":
  fullprompt += scriptPrompt.replace("[language]","Python");
  defaultTemperature = 0;
  break;  
case "SJ":
  fullprompt += scriptPrompt.replace("[language]","Javascript");
  defaultTemperature = 0;
  break;    
default:
  fullprompt += `
  -Return answer in markdown format
  -You are tasked with the following: ${userInput}`;
}
if (lengthHash === 'S') {
fullprompt += `
  -Make the text a bit more concise.`;
}
if (lengthHash === 'L') {
fullprompt += `
  -Make the text a bit longer.`;
}
fullprompt += `
######`;
return fullprompt;
};
function setTemperature(temp, defaultTemp) {
  switch (temp) {
    case "---":
      return 0;
    case "--":
      return 0.2;
    case "-":
      return 0.4;
    case "+":
      return 0.6;
    case "++":
      return 0.8;
    case "+++":
      return 1;
    default:
      return defaultTemp;
    }
};

//################
// Options Template
//################
// @TODO: styling
const options = `
* [Retry](submit:retry) - Rerun generation with option to update prompt
* [Followup](submit:followup) - Followup Question this response
* [Edit](submit:edit) - Edit response in editor
* [Copy](submit:copy) - Copy response to clipboard
* [Paste](submit:paste) - Paste response into highlighted text
`;

//################
// Main Function
//################
/**
 *
 * @param {*} prompt
 * @param {*} humanChatMessage
 */
async function promptAgainstHighlightedText(
  prompt = formatPrompt(userSystemInput),
  humanChatMessage = userPrompt
) {
  //#########
  // Helpers
  //########
  // exit script on cancel
  const cancelChat = () => {
    process.exit(1);
  };

  /**
   * Paste text to highlighted text and exit script
   * @param {*} text
   */
  const pasteTextAndExit = async (text) => {
    await setSelectedText(text);
    process.exit(1);
  };

  /**
   * Copy text to clipboard and exit script
   * @param {*} text
   */
  const copyToClipboardAndExit = async (text) => {
    await clipboard.writeText(currentMessage);
    process.exit(1);
  };

  let currentMessage = "";
  const llm = new ChatOpenAI({
    // 0 = "precise", 1 = "creative"
    temperature: setTemperature(temperatureHash, defaultTemperature),
    // modelName: "gpt-4", // uncomment to use GPT-4 (requires beta access)
    openAIApiKey: openAIApiKey,
    // turn off to only get output when the AI is done
    streaming: true,
    callbacks: [
      {
        handleLLMStart: async () => {
          log(`handleLLMStart`);
          toast(`Generating...`, { duration: 1000 });
          // render initial message
        },
        handleLLMNewToken: async (token) => {
          log(`handleLLMNewToken`);
          // each new token is appended to the current message
          // and then rendered to the screen
          currentMessage += token;
          // render current message
          await div({
            html: md(currentMessage),
            // @TODO: Figure out how to get ESC to trigger a cancel
            onAbandon: cancelChat,
            onEscape: cancelChat,
            onBackspace: cancelChat,
            // if this is set to false you can click outside the window to cancel
            // which works, but would be nice to also have ESC work
            ignoreBlur: false,
            focus: true,
            // hint: `Press ESC to cancel`,
          });
        },
        handleLLMError: async (err) => {
          dev({ err });
        },
        handleLLMEnd: async () => {
          log(`used prompt: ${prompt}`);
          log(`used temperature: ${setTemperature(temperatureHash, defaultTemperature)}`);
          log(`used text to handle: ${userPrompt}`);
          log(`handleLLMEnd`);
          // render final message with options
          let html = md(currentMessage + options);
          // wait for user to select an option
          const selectedOption = await div(html, {
            ignoreBlur: true,
            focus: true,
            // have paste on text on submit?
            // onSubmit: () => pasteTextAndExit(currentMessage),
          });
          // handle selected option
          switch (selectedOption) {
            case "paste":
              await pasteTextAndExit(currentMessage);
            case "retry":
              // reset current message
              currentMessage = "";
              // prompt again with new prompt
              // press enter to use original prompt
              const followUp = await arg({
                placeholder: userSystemInput,
                hint: "Press enter to use the same prompt or " + promptHint,
              });
              prompt = formatPrompt(followUp);
              await processMessage(followUp);
              break;
            case "followup":
              // keep current message (response from AI) and set it as the new input text
              userPrompt = currentMessage;
              // prompt again with new prompt on this response
              const question = await arg({
                placeholder: "",
                hint: promptHint,
              });
              prompt = formatPrompt(question);
              await processMessage(question);
              break;  
            case "edit":
              // @TODO still need to figure out best way to handle submit and abort
              // would like custom buttons for triggering all of the actions like copy, paste, etc
              await editor({
                value: currentMessage,
                onEscape: async (state) => await copyToClipboardAndExit(state),
                onSubmit: async (state) => await pasteTextAndExit(state),
              });
              break;
            case "copy":
              await copyToClipboardAndExit(currentMessage);
            case "save":
              await inspect(currentMessage, `/conversations/${Date.now()}.md`);
              exitChat();
            default:
              copyToClipboardAndExit(currentMessage);
          }
          await optionHandler(selectedOption);
        },
      },
    ],
  });
  //###########
  // Main Loop
  //###########
  // runs the language model until the user cancels
  while (true) {
    await llm.call([
      new SystemChatMessage(prompt),
      new HumanChatMessage(humanChatMessage),
    ]);
  }
}

promptAgainstHighlightedText();
`

**Comment by stevedubois** [comment]
I better attach the file
[prompt-anywhere.txt](https://github.com/johnlindquist/kit/files/11414812/prompt-anywhere.txt)

**Reply by mabry1985** [reply]
very cool @stevedubois thanks for sharing!

**Comment by stevedubois** [comment]
Hello Josh,

I have an additional question. I attempted to test your script with GPT-4, but when I uncommented the following line:
// modelName: "gpt-4", // uncomment to use GPT-4 (requires beta access),
the script ceased to function. Despite having an API key and a GPT Plus subscription, I am unsure of what I am missing.



Van: Josh Mabry ***@***.***>
Datum: maandag, 8 mei 2023 om 19:28
Aan: johnlindquist/kit ***@***.***>
CC: Dubois Steve ***@***.***>, Mention ***@***.***>
Onderwerp: Re: [johnlindquist/kit] Prompt ChatGPT from anywhere you can input text (Discussion #1220)

very cool @stevedubois<https://github.com/stevedubois> thanks for sharing!

**Comment by mabry1985** [comment]
The GPT4 api access is separate from ChatGPT plus. You need to signup for
beta access
https://openai.com/waitlist/gpt-4-api

On Wed, May 24, 2023 at 1:43 AM stevedubois ***@***.***>
wrote:

> Hello Josh,
>
> I have an additional question. I attempted to test your script with GPT-4,
> but when I uncommented the following line:
> // modelName: "gpt-4", // uncomment to use GPT-4 (requires beta access),
> the script ceased to function. Despite having an API key and a GPT Plus
> subscription, I am unsure of what I am missing.
>
>
>
> Van: Josh Mabry ***@***.***>
> Datum: maandag, 8 mei 2023 om 19:28
> Aan: johnlindquist/kit ***@***.***>
> CC: Dubois Steve ***@***.***>, Mention ***@***.***>
> Onderwerp: Re: [johnlindquist/kit] Prompt ChatGPT from anywhere you can
> input text (Discussion #1220)
>
> very cool @stevedubois<https://github.com/stevedubois> thanks for sharing!
>
> —
> Reply to this email directly, view it on GitHub<
> https://github.com/johnlindquist/kit/discussions/1220#discussioncomment-5840733>,
> or unsubscribe<
> https://github.com/notifications/unsubscribe-auth/ABK3NDSTPYWGMAIU2KR7U7DXFEUJXANCNFSM6AAAAAAXP3BFFA
> >.
> You are receiving this because you were mentioned.Message ID: ***@***.***>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/johnlindquist/kit/discussions/1220#discussioncomment-5986812>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHQZCX6NA5EBV6ZGF4XEQILXHXC4PANCNFSM6AAAAAAXP3BFFA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Josh

**Comment by stevedubois** [comment]
Thanks!

I didn’t know.
Filled in the form.


Met vriendelijke groeten

Steve Dubois
Cel ICT
Stedelijk Onderwijs Gent | Departement Onderwijs, Opvoeding & Jeugd | Stad Gent
Bezoekadres | Woodrow Wilsonplein 1, 9000 Gent
Postadres | Stadhuis, Botermarkt 1, 9000 Gent
Schoolbord.gent.be<https://schoolbord.gent.be/>
Denk aan het milieu voor u mails print.
Aansprakelijkheidsbeperker: Deze e-mail en bijlagen kunnen informatie bevatten die vertrouwelijk is en/of beschermd wordt door intellectuele eigendomsrechten. Lees meer ...<http://www.gent.be/gent.htm?id=131587>
[Image]<https://mia.gent.be/dw_site/3f579fabeeb8aa4f4df4b26330a5c9a6.php>


Van: Josh Mabry ***@***.***>
Datum: woensdag, 24 mei 2023 om 16:47
Aan: johnlindquist/kit ***@***.***>
CC: Dubois Steve ***@***.***>, Mention ***@***.***>
Onderwerp: Re: [johnlindquist/kit] Prompt ChatGPT from anywhere you can input text (Discussion #1220)
The GPT4 api access is separate from ChatGPT plus. You need to signup for
beta access
https://openai.com/waitlist/gpt-4-api

On Wed, May 24, 2023 at 1:43 AM stevedubois ***@***.***>
wrote:

> Hello Josh,
>
> I have an additional question. I attempted to test your script with GPT-4,
> but when I uncommented the following line:
> // modelName: "gpt-4", // uncomment to use GPT-4 (requires beta access),
> the script ceased to function. Despite having an API key and a GPT Plus
> subscription, I am unsure of what I am missing.
>
>
>
> Van: Josh Mabry ***@***.***>
> Datum: maandag, 8 mei 2023 om 19:28
> Aan: johnlindquist/kit ***@***.***>
> CC: Dubois Steve ***@***.***>, Mention ***@***.***>
> Onderwerp: Re: [johnlindquist/kit] Prompt ChatGPT from anywhere you can
> input text (Discussion #1220)
>
> very cool @stevedubois<https://github.com/stevedubois> thanks for sharing!
>
> —
> Reply to this email directly, view it on GitHub<
> https://github.com/johnlindquist/kit/discussions/1220#discussioncomment-5840733>,
> or unsubscribe<
> https://github.com/notifications/unsubscribe-auth/ABK3NDSTPYWGMAIU2KR7U7DXFEUJXANCNFSM6AAAAAAXP3BFFA
> >.
> You are receiving this because you were mentioned.Message ID: ***@***.***>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/johnlindquist/kit/discussions/1220#discussioncomment-5986812>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHQZCX6NA5EBV6ZGF4XEQILXHXC4PANCNFSM6AAAAAAXP3BFFA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
--
Josh
